### PR TITLE
GS-HW: Use correct alphas when AA1 is enabled + ABE disabled.

### LINKED ISSUE #983: Final Fantasy X core (?) bug in Kilika
There's a small issue that I've noticed in Kilika. If you look closely at the provided screenshot, there's a part of untextured terrain showing up in there. It looks blue and it's down and to the right of Tidus. This doesn't happen on the PS2. I think it's a core bug because it happens on all of GSdx's renderers (D3D9, D3D11, OGL...hardware and software), as well as ZeroGS. It also happens on PCSX2 versions going as far back as 0.9.6.

At first I thought it was an issue with the widescreen patch since I've seen the issue happen before but on parts of the map that were clearly never supposed to be shown in a 4:3 aspect ratio (http://i.imgur.com/5CZ4nCC.png VS http://i.imgur.com/GH8i8Rn.png) so I turned off widescreen patches just in case.

Here's what it looks like in PCSX2:
si1qet4

Here's what it's supposed to look like:

Just for the lulz, here's what it looks like on PCSX2 0.9.6 using ZeroGS:
fxi9xaz

### LINKED ISSUE #3341: Final Fantasy X Glitched Level Meshes
***PCSX2 version:***
1.5.0-dev-3406

***PCSX2 options:***
Default

***Plugins used:***
Default

***Description of the issue:***
Some parts of the "Macalania Woods - Lake Road" location have "holes" in them.
Similar to #983

***PCSX2:*** 

tsehtrs

***PS2:*** 

redhg

***Last known version to work:***

none known, goes far back as 0.9.8

### LINKED ISSUE #6644: [BUG]: Bakusou Dekotora Densetsu HW rendering issues
### Describe the Bug

as mentioned on #6493 the game has issues in  HW rendering modes 
image

SW mode 

image

### Reproduction Steps

play said game 

### Expected Behavior

no HW rendering issues 

### PCSX2 Revision

1.7.3080

### Operating System

Linux (64bit) - Specify Distro Below

### If Linux - Specify Distro

Ubuntu 20.04

### CPU

i5 3210m 

### GPU

intel hd 4000 mobile 

### GS Settings

_No response_

### Emulation Settings

_No response_

### GS Window Screenshots

_No response_

### Logs & Dumps

[gs_20220620141034_Bakusou.Dekotora.Densetsu.-.Otoko.Hanamichi.Yume.Roman_SLPM-65234.zip](https://github.com/PCSX2/pcsx2/files/9125593/gs_20220620141034_Bakusou.Dekotora.Densetsu.-.Otoko.Hanamichi.Yume.Roman_SLPM-65234.zip)

### PR DESCRIPTION
**Pull request by refractionpcsx2**
### Description of Changes
Correctly handles the blending for the body of triangles etc when Edge Antialiasing is enabled.

### Rationale behind Changes
When Edge Anti-aliasing is enabled and alpha blending is disabled, there is still a blend effect, but the alpha outputted and used for the blend is 128 (1.0), before we just ignored this completely causing incorrect data to be written. Also the Alpha Test relies on the pixel alpha being the coverage sample, otherwise it can be erroneously culled. Behaviour matches software mode in terms of written alpha and test alpha.

### Suggested Testing Steps
Test games, especially those that are known to use AA, make sure nothing is broken.

⚠️ NOTE: This will NOT make AA work in hardware mode, if edges are supposed to get anti-aliased, they still won't, this just improves the behaviour on handling the majority of the triangles being drawn. ⚠️ 

Fixes a few things listed in https://github.com/PCSX2/pcsx2/issues/5069
Fixes #3341 and Fixes #983 (Final Fantasy X holes in terrain)
Fixes Trapt - holes in the terrain/black patches (some floor still a problem, but this also affects SW mode).
Fixes #6644 - Bakusou Dekotora Densetsu missing road.

FFX:

Before
image
After
image

Bakusou Dekotora:

Before
image

After
image

Trapt:

Before
image

After
image

### CODE REVIEW

**Comment by refractionpcsx2** [comment]
@tellowkrinkle can you make sure I didn't cock up Metal?

There's a GS dump for FFX here if you need it https://github.com/PCSX2/pcsx2/issues/3341#issuecomment-622210400

**Comment by tellowkrinkle** [comment]
> @tellowkrinkle can you make sure I didn't cock up Metal?

Seems to work fine

Though I'd prefer if you didn't check booleans with `PS_FIXED_ONE_A == 1`
```
if (PS_FIXED_ONE_A)
	C.a = 128.0f;
```

**Comment by refractionpcsx2** [comment]
> > @tellowkrinkle can you make sure I didn't cock up Metal?
> 
> Seems to work fine
> 
> Though I'd prefer if you didn't check booleans with `PS_FIXED_ONE_A == 1`
> 
> ```
> if (PS_FIXED_ONE_A)
> 	C.a = 128.0f;
> ```

yep, sorry, laps in brain. Fixed.

**Review by tellowkrinkle — APPROVED** [review]

File: pcsx2/GS/Renderers/HW/GSRendererHW.cpp [hunk 1]
```diff
@@ -2336,13 +2375,13 @@ void GSRendererHW::EmulateBlending(bool& DATE_PRIMID, bool& DATE_BARRIER, bool&
...
 	// it requires coverage sample so it's safer to turn it off instead.
-	const bool AA1 = PRIM->AA1 && (m_vt.m_primclass == GS_LINE_CLASS);
+	const bool AA1 = PRIM->AA1 && (m_vt.m_primclass == GS_LINE_CLASS || m_vt.m_primclass == GS_TRIANGLE_CLASS);
 	// PABE: Check condition early as an optimization.
 	const bool PABE = PRIM->ABE && m_env.PABE.PABE && (GetAlphaMinMax().max < 128);
 	// FBMASK: Color is not written, no need to do blending.
 	const u32 temp_fbmask = m_conf.ps.dfmt == 2 ? 0x00F8F8F8 : 0x00FFFFFF;
 	const bool FBMASK = (m_context->FRAME.FBMSK & temp_fbmask) == temp_fbmask;
-
+	const bool force_a_one = !PRIM->ABE && AA1;
```
Comment (lightningterror) [inline pcsx2/GS/Renderers/HW/GSRendererHW.cpp:2384]: I'd move this where the if is, just above it.

File: pcsx2/GS/Renderers/HW/GSRendererHW.cpp [hunk 2]
```diff
@@ -2336,13 +2375,13 @@ void GSRendererHW::EmulateBlending(bool& DATE_PRIMID, bool& DATE_BARRIER, bool&
...
-	const bool AA1 = PRIM->AA1 && (m_vt.m_primclass == GS_LINE_CLASS);
+	const bool AA1 = PRIM->AA1 && (m_vt.m_primclass == GS_LINE_CLASS || m_vt.m_primclass == GS_TRIANGLE_CLASS);
 	// PABE: Check condition early as an optimization.
 	const bool PABE = PRIM->ABE && m_env.PABE.PABE && (GetAlphaMinMax().max < 128);
 	// FBMASK: Color is not written, no need to do blending.
 	const u32 temp_fbmask = m_conf.ps.dfmt == 2 ? 0x00F8F8F8 : 0x00FFFFFF;
 	const bool FBMASK = (m_context->FRAME.FBMSK & temp_fbmask) == temp_fbmask;
-
+	
 	// No blending or coverage anti-aliasing so early exit
```
Comment (lightningterror) [inline pcsx2/GS/Renderers/HW/GSRendererHW.cpp:2385]: Remove the tab.

File: pcsx2/GS/Renderers/HW/GSRendererHW.cpp [hunk 3]
```diff
@@ -2360,11 +2399,20 @@ void GSRendererHW::EmulateBlending(bool& DATE_PRIMID, bool& DATE_BARRIER, bool&
 	m_conf.ps.blend_b = ALPHA.B;
 	m_conf.ps.blend_c = ALPHA.C;
 	m_conf.ps.blend_d = ALPHA.D;
+	const bool force_a_one = !PRIM->ABE && AA1;
```
Comment (lightningterror) [inline pcsx2/GS/Renderers/HW/GSRendererHW.cpp:2402]: I'd move it after the comment, also maybe replace the comment with AA1, and move the old one in the function.

**Review by lightningterror — APPROVED** [review]

**Review by Mrlinkwii — APPROVED** [review]

**Pull request merged**